source/internal/_toString.js

```diff
@@ -26,10 +26,14 @@ export default function _toString(x, seen) {
       return typeof x === 'object' ? 'new Boolean(' + recur(x.valueOf()) + ')' : x.toString();
     case '[object Date]':
       return 'new Date(' + (isNaN(x.valueOf()) ? recur(NaN) : _quote(_toISOString(x))) + ')';
+    case '[object Map]':
+      return 'new Map(' + recur(Array.from(x)) + ')';
     case '[object Null]':
       return 'null';
     case '[object Number]':
       return typeof x === 'object' ? 'new Number(' + recur(x.valueOf()) + ')' : 1 / x === -Infinity ? '-0' : x.toString(10);
+    case '[object Set]':
+      return 'new Set(' + recur(Array.from(x).sort()) + ')';
```

Comment:
Why are we sorting the keys here? If key order matters then we should maintain the current ordering. If it doesn't matter then there's no reason to change it.

Reply:
> If key order matters

It doesn't. Sets are by definition unordered.

Sorting guarantees that if `equals(setA, setB)`, then `eqBy(toString, setA, setB)`.